cmd/kubeadm/app/master/manifests_test.go

```diff
@@ -0,0 +1,577 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package master
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
+	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
+	api "k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/pkg/util/intstr"
+)
+
+func TestWriteStaticPodManifests(t *testing.T) {
+	tmpdir, err := ioutil.TempDir("", "")
+	if err != nil {
+		t.Fatalf("Couldn't create tmpdir")
+	}
+	defer os.Remove(tmpdir)
+
+	// set up tmp GlobalEnvParams values for testing
+	oldEnv := kubeadmapi.GlobalEnvParams
+	kubeadmapi.GlobalEnvParams.KubernetesDir = fmt.Sprintf("%s/etc/kubernetes", tmpdir)
+	defer func() { kubeadmapi.GlobalEnvParams = oldEnv }()
+
+	var tests = []struct {
+		cfg      *kubeadmapi.MasterConfiguration
+		expected bool
+	}{
+		{
+			cfg:      &kubeadmapi.MasterConfiguration{},
+			expected: true,
+		},
+	}
+	for _, rt := range tests {
+		actual := WriteStaticPodManifests(rt.cfg)
+		if (actual == nil) != rt.expected {
+			t.Errorf(
+				"failed WriteStaticPodManifests with an error:\n\texpected: %t\n\t  actual: %t",
+				rt.expected,
+				(actual == nil),
+			)
+		}
+	}
+}
+
+func TestEtcdVolume(t *testing.T) {
+	var tests = []struct {
+		cfg      *kubeadmapi.MasterConfiguration
+		expected api.Volume
+	}{
+		{
+			cfg: &kubeadmapi.MasterConfiguration{},
+			expected: api.Volume{
+				Name: "etcd",
+				VolumeSource: api.VolumeSource{
+					HostPath: &api.HostPathVolumeSource{
+						Path: kubeadmapi.GlobalEnvParams.HostEtcdPath},
+				}},
+		},
+	}
+
+	for _, rt := range tests {
+		actual := etcdVolume(rt.cfg)
+		if actual.Name != rt.expected.Name {
+			t.Errorf(
+				"failed etcdVolume:\n\texpected: %s\n\t  actual: %s",
+				rt.expected.Name,
+				actual.Name,
+			)
+		}
+		if actual.VolumeSource.HostPath.Path != rt.expected.VolumeSource.HostPath.Path {
+			t.Errorf(
+				"failed etcdVolume:\n\texpected: %s\n\t  actual: %s",
+				rt.expected.VolumeSource.HostPath.Path,
+				actual.VolumeSource.HostPath.Path,
+			)
+		}
+	}
+}
+
+func TestEtcdVolumeMount(t *testing.T) {
+	var tests = []struct {
+		expected api.VolumeMount
+	}{
+		{
+			expected: api.VolumeMount{
+				Name:      "etcd",
+				MountPath: "/var/etcd",
```

Comment:
Its to make sure that nothing gets changed in that function.  I think its a small and simple tests that will ensure that if something got changed this would catch it. Can't see the harm in it.